src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,25 @@
+package duke.command;
+
+import duke.storage.Storage;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+public abstract class Command {
+
+    /**
+     * Executes the command.
+     *
+     * @param tasks   The TaskList containing all existing tasks.
+     * @param ui      The Ui for printing purposes.
+     * @param storage The Storage for saving tasks to file.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage);
+
+    /**
+     * Checks if it is a bye command.
+     *
+     * @return A boolean: true if it is a bye command.
+     */
+    public abstract boolean isExit();
```

Comment:
Consider an alternative implementation: you can implement `isExit` as follows:
```Java
public boolean isExit() {
    return this instanceof ExitCommand;
}
```
As a result, you don't need to reimplement this method in all its child classes.